pkg/grpc/request_headers_authenticator.go

```diff
@@ -0,0 +1,13 @@
+package grpc
+
+import (
+	"context"
+
+	"github.com/buildbarn/bb-storage/pkg/auth"
+)
+
+// RequestHeadersAuthenticator can be used to grant or deny access to a server
+// based on request headers, typically from an HTTP or gRPC request.
+type RequestHeadersAuthenticator interface {
```

Comment:
I think this should live in pkg/auth, as it's not specific to any specific transport protocol, right? Just ones that "provide string key/value headers".